### PR TITLE
fix: make torch.load safer with weights_only=True everywhere possible

### LINKED ISSUE #621: Investigate whether weights_only=True works with all of our torch.load instances
### Bug description

Prep :

`Normalizing pitch values:  0%|     | 0/999 [00:00<?, ?it/s]/gpfs/fs5/nrc/nrc-fs1/ict/others/u/tes001/TxT2SPEECH/EveryVoice_PT_2.4.1/everyvoice/preprocessor/preprocessor.py:440: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
 pitch = torch.load(path)
[11:16](https://nrcilt.slack.com/archives/C058A7R19HN/p1736795786403849)`

When we start training a FP. we get these warnings  Once:

`/gpfs/fs5/nrc/nrc-fs1/ict/others/u/tes001/TxT2SPEECH/EveryVoice_PT_latest_manual/everyvoice/dataloader/__init__.py:46: FutureWarning: You are using `torch.load` with `weights_only=Fals
e` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (
See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. Th
is limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the u
ser via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an
 issue on GitHub for any issues related to this experimental feature.
 self.train_dataset = torch.load(self.train_path)`

`/gpfs/fs5/nrc/nrc-fs1/ict/others/u/tes001/TxT2SPEECH/EveryVoice_PT_latest_manual/everyvoice/dataloader/__init__.py:47: FutureWarning: You are using `torch.load` with `weights_only=Fals
e` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (
See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. Th
is limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the u
ser via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an
 issue on GitHub for any issues related to this experimental feature.
 self.val_dataset = torch.load(self.val_path)`

Every epoch afterwards:

`/gpfs/fs5/nrc/nrc-fs1/ict/others/u/tes001/TxT2SPEECH/EveryVoice_PT_latest_manual/everyvoice/model/feature_prediction/FastSpeech2_lightning/fs2/model.py:435: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.`

### How to reproduce the bug

```python
install torch==2.4.1 and run preprocesssing and train.
```

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>
  ```
  #- EveryVoice Diagnostic information
  #- EveryVoice version: 
  #- Python version: 
  #- System: 
  ```
</details>

### More info

_No response_

### PR DESCRIPTION
**Pull request by joanise**
### PR Goal? 

Prepare EV to be compatible with pytorch 2.4.1 and future, when the default `weights_only=False` eventually will get changed to `weights_only=True`.

https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models documents how one should not load data files received from a third party with `weights_only=False`, so I figure it's important to at least change how we load our checkpoints.

To create this PR, I set `weigths_only=True` on every single `torch.load()` call in EV, and ran stuff (using unit testing and the wip regress branch): it turns out our checkpoints and most of our uses are OK with only "tensors, primitive types, dictionaries". The only exception I found is prepared datasets, and I think it's reasonable to use weights in that case since those files will be created locally by the user.

Ideally, we might eventually try to make those work with weights only as well, but I've played a bit with that and the way things are we would have to call [torch.serialization.add_safe_globals()](https://pytorch.org/docs/stable/notes/serialization.html#torch.serialization.add_safe_globals) with about a dozen classes we pickle in our prepared datasets.

### Fixes? 

Fixes #621

### Priority? 

I would say beta

### Tests added? 

partly covered by existing CI, will be more covered by the regression suite.

### How to test? 

Run all the train and synthesize commands. The easiest way at the moment is to get the regression test from dev.ej/regression and run it on this branch of the code.

### Confidence? 

high

### Version change? 

no

### Related PRs? 

https://github.com/EveryVoiceTTS/HiFiGAN_iSTFT_lightning/pull/46
https://github.com/EveryVoiceTTS/FastSpeech2_lightning/pull/107
https://github.com/EveryVoiceTTS/DeepForcedAligner/pull/32

### CODE REVIEW

**Review by marctessier — APPROVED** [review]
I ran some test using PT 2.1 and also with  2.4.1 for fun to see if the message goes away and it did :-)  Training worked , synth etc..
